File: library/src/main/java/com/getstream/sdk/chat/notifications/DeviceRegisteredListener.java
package com.getstream.sdk.chat.notifications;

import org.jetbrains.annotations.NotNull;

public interface DeviceRegisteredListener {

    /**
     * Callback called when device registered on server successfully
     */
    void onDeviceRegisteredSuccess();

    /**
     * Callback called when we can't register device on server
     *
     * @param errorMessage - Message from server
     * @param errorCode    - error code from server
     */
    void onDeviceRegisteredError(@NotNull String errorMessage, int errorCode);
}

File: app/src/main/java/io/getstream/chat/android/core/poc/library/notifications/DeviceRegisteredListener.kt
package io.getstream.chat.android.core.poc.library.notifications

import io.getstream.chat.android.core.poc.library.errors.ChatError

interface DeviceRegisteredListener {
    /**
     * Callback called when device registered on server successfully
     */
    fun onDeviceRegisteredSuccess()

    /**
     * Callback called when we can't register device on server
     *
     * @param errorMessage - Message from server
     * @param errorCode    - error code from server
     */
    fun onDeviceRegisteredError(error: ChatError)
}